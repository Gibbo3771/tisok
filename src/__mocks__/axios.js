const slips = [
  { slip: { advice: "life is good", slip_id: 1 } },
  { slip: { advice: "apples taste good", slip_id: 2 } },
  { slip: { advice: "death is certain, get over it", slip_id: 3 } }
];

module.exports = {
  get: id => {
    return Promise.resolve({
      data: {
        slip: {
          advice: "testing is important",
          slip_id: 1
        }
      }
    });
  }
};
