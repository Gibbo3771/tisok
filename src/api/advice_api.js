// @flow
import axios from "axios";

const URL = "https://api.adviceslip.com/advice";

class AdviceAPI {
  random = () => {
    return axios.get<string, any>(URL);
  };

  get = (id: number) => {
    return axios.get<string, any>(`${URL}/${id}`);
  };
}

export default new AdviceAPI();
