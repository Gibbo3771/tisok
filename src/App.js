import React from "react";
import { BrowserRouter as Router, Route } from "react-router-dom";
import AdviceSlip from "./componets/AdviceSlip";
import AdviceButton from "./componets/AdviceButton";

import { random, get } from "./api/advice_api";

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      advice: null
    };
  }

  render() {
    const { advice } = this.state;
    return (
      <Router>
        <Route
          path="/:id?"
          render={props => (
            <div className="grid">
              <AdviceSlip {...props} advice={advice} onReady={this.getAdvice} />
              <AdviceButton onClick={this.handleClick} />
            </div>
          )}
        />
      </Router>
    );
  }

  handleClick = () => {
    this.getRandomAdvice();
  };

  splitAdvice = advice => {
    this.setState({ advice: "" });
    this.setState({ advice: advice.split("") });
  };

  getAdvice = id => {
    if (!id) this.getRandomAdvice();
    else this.getAdviceByID(id);
  };

  getRandomAdvice = () => {
    random()
      .then(response => {
        this.splitAdvice(response.data.slip.advice);
      })
      .catch(err => console.log(err));
  };

  getAdviceByID = id => {
    get(id)
      .then(response => {
        console.log(response);

        this.splitAdvice(response.data.slip.advice);
      })
      .catch(err => console.log(err));
  };
}
