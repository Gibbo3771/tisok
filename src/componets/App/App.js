// @flow
import React from "react";
import { BrowserRouter as Router, Route } from "react-router-dom";
import AdviceSlip from "../AdviceSlip/AdviceSlip";
import AdviceButton from "../AdviceButton/AdviceButton";
import ShareModal from "../ShareModal/ShareModal";

import api from "../../api/advice_api";
import SocialMediaPanel from "../SocialMediaPanel/SocialMediaPanel";

export type Props = {};

export type State = {
  slip: any,
  url: string
};

export default class App extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      slip: {},
      url: window.location.href
    };
  }

  render() {
    const { slip, url } = this.state;
    return (
      <Router>
        <Route
          path="/:id?"
          render={props => (
            <div className="grid">
              <AdviceSlip
                {...props}
                slip={slip}
                onReady={id => this.getAdvice(id)}
              />
              <AdviceButton {...props} onClick={() => this.handleClick()} />
              <SocialMediaPanel url={url} />
            </div>
          )}
        />
      </Router>
    );
  }

  handleClick = () => {
    this.getRandomAdvice();
  };

  getAdvice = (id?: number) => {
    if (!id) this.getRandomAdvice();
    else this.getAdviceByID(id);
  };

  setAdvice = (response: any) => {
    const { slip } = response.data;
    this.setState({ slip: {} });
    this.setState({
      slip: slip,
      url: `${window.location.href}${slip.slip_id}`
    });
  };

  getRandomAdvice = () => {
    return api
      .random()
      .then(response => {
        this.setAdvice(response);
      })
      .catch(err => console.log(err));
  };

  getAdviceByID = (id: number) => {
    api
      .get(id)
      .then(response => {
        this.setAdvice(response);
      })
      .catch(err => console.log(err));
  };
}
