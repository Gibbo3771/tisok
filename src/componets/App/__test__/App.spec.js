// @flow
import React from "react";
import App from "../App";
import axios from "axios";
import api from "../../../api/advice_api";

jest.mock("axios");

let wrapper = mount(<App />);
let instance = wrapper.instance();

describe("App", () => {
  beforeEach(() => {
    wrapper = mount(<App />);
    instance = wrapper.instance();
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have valid initial state", () => {
    expect(wrapper.state().slip).toBeDefined();
    expect(wrapper.state().url).toBeDefined();
  });
  it("responds to the advice button being clicked", async () => {
    const clickSpy = sinon.spy(instance, "handleClick");
    const getRandomAdvice = sinon.spy(instance, "getRandomAdvice");
    const axiosSpy = sinon.spy(axios, "get");
    expect(wrapper.exists("#advice-button")).toBeTruthy();
    wrapper.find("#advice-button").simulate("click");
    expect(clickSpy.calledOnce).toBeTruthy();
    expect(getRandomAdvice.callCount).toEqual(1);
    expect(axiosSpy.callCount).toEqual(1);
    axiosSpy.restore();
  });
  it("sets the slip to new values with button clicks", () => {
    expect(wrapper.exists("#advice-button")).toBeTruthy();
    wrapper.find("#advice-button").simulate("click");
    expect(wrapper.state().slip.slip_id).toEqual(1);
    expect(wrapper.state().slip.advice).toEqual("testing is important");
  });
  it("sets the slip when setAdvice is called", () => {
    const response = {
      data: { slip: { slip_id: 5, advice: "solid advice" } }
    };
    instance.setAdvice(response);
    expect(wrapper.state().slip.slip_id).toEqual(5);
    expect(wrapper.state().slip.advice).toEqual("solid advice");
  });
  it("sets the slip when getRandomAdvice is called", () => {
    const spy = sinon.spy(axios, "get");
    instance.getRandomAdvice();
    expect(wrapper.state().slip.slip_id).toEqual(1);
    expect(wrapper.state().slip.advice).toEqual("testing is important");
    expect(spy.callCount).toEqual(1);
    spy.restore();
  });
  it("sets the slip when getAdviceByID is called", () => {
    const spy = sinon.spy(axios, "get");
    instance.getAdviceByID(1);
    expect(wrapper.state().slip.slip_id).toEqual(1);
    expect(wrapper.state().slip.advice).toEqual("testing is important");
    expect(spy.callCount).toEqual(1);
    spy.restore();
  });
  it("has a social media panel", () => {
    expect(wrapper.exists(".social-media-panel")).toBeTruthy();
  });
});
