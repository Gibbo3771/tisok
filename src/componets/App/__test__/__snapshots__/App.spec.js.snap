// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<App>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Route
        path="/:id?"
        render={[Function]}
      >
        <div
          className="grid"
        >
          <AdviceSlip
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {
                  "id": undefined,
                },
                "path": "/:id?",
                "url": "/",
              }
            }
            onReady={[Function]}
            slip={Object {}}
          >
            <div />
          </AdviceSlip>
          <AdviceButton
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {
                  "id": undefined,
                },
                "path": "/:id?",
                "url": "/",
              }
            }
            onClick={[Function]}
          >
            <button
              className="advice-get"
              id="advice-button"
              onClick={[Function]}
            >
              (｡^‿^｡)
            </button>
          </AdviceButton>
          <SocialMediaPanel
            url="http://localhost/"
          >
            <div
              className="social-media-panel"
              data-for="social-media-tooltip"
              data-tip="Share"
            >
              <CreatedButton
                className="share-twitter"
                hashtags={
                  Array [
                    "tisok",
                    "lifeprotips",
                  ]
                }
                title="Advice to live by..."
                url="http://localhost/"
                windowHeight={400}
                windowWidth={550}
              >
                <ShareButton
                  className="share-twitter"
                  disabledStyle={
                    Object {
                      "opacity": 0.6,
                    }
                  }
                  hashtags={
                    Array [
                      "tisok",
                      "lifeprotips",
                    ]
                  }
                  network="twitter"
                  networkLink={[Function]}
                  openWindow={true}
                  opts={
                    Object {
                      "hashtags": Array [
                        "tisok",
                        "lifeprotips",
                      ],
                      "title": "Advice to live by...",
                      "via": undefined,
                    }
                  }
                  role="button"
                  tabIndex="0"
                  title="Advice to live by..."
                  url="http://localhost/"
                  windowHeight={400}
                  windowWidth={550}
                >
                  <div
                    className="SocialMediaShareButton SocialMediaShareButton--twitter share-twitter"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    role="button"
                    style={Object {}}
                    tabIndex="0"
                  >
                    <i
                      className="fab fa-twitter-square icon-twitter icon"
                    />
                  </div>
                </ShareButton>
              </CreatedButton>
              <ReactTooltip
                className="tooltip"
                clickable={false}
                effect="solid"
                id="social-media-tooltip"
                insecure={true}
                place="top"
                resizeHide={true}
                type="tooltip"
                wrapper="div"
              >
                <div
                  className="__react_component_tooltip place-top type-dark "
                  data-id="tooltip"
                  id="social-media-tooltip"
                />
              </ReactTooltip>
            </div>
          </SocialMediaPanel>
        </div>
      </Route>
    </Router>
  </BrowserRouter>
</App>
`;
